sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py

```diff
@@ -1631,3 +1631,108 @@ def __init__(
             return await self._client.append_blob.append_block(**options)  # type: ignore
         except StorageErrorException as error:
             process_storage_error(error)
+
+    @distributed_trace_async()
+    async def append_block_from_url(self, copy_source_url,  # type: str
+                                    source_range_start=None,  # type Optional[int]
+                                    source_range_end=None,  # type Optional[int]
+                                    source_content_md5=None,  # type: Optional[bytearray]
+                                    maxsize_condition=None,  # type: Optional[int]
+                                    appendpos_condition=None,  # type: Optional[int]
+                                    lease=None, if_modified_since=None,
+                                    if_unmodified_since=None, if_match=None,
+                                    if_none_match=None, source_if_modified_since=None,
+                                    source_if_unmodified_since=None, source_if_match=None,
+                                    source_if_none_match=None, timeout=None):
```

Comment:
Could we move some of these parameters into `**kwargs`?
I think everything after (and including) `lease` could be simplified with `**kwargs`.
To see an example of this, check the copy API here:
https://github.com/Azure/azure-sdk-for-python/blob/feature/storage-preview3/sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py#L1255

You can leave all the options in the docstring :)